src/ott/neural/solvers/flows.py

```diff
@@ -0,0 +1,250 @@
+      :math:`x_1` at time :math:`t`.
+
+    Args:
+      t: Time :math:`t`.
+      x_0: Sample from the source distribution.
+      x_1: Sample from the target distribution.
+    """
+    pass
+
+  def compute_xt(
+      self, noise: jnp.ndarray, t: jnp.ndarray, x_0: jnp.ndarray,
+      x_1: jnp.ndarray
+  ) -> jnp.ndarray:
+    """Sample from the probability path.
+
+      Sample from the probability path between :math:`x_0` and :math:`x_1` at
+      time :math:`t`.
+
+    Args:
+      noise: Noise sampled from a standard normal distribution.
+      t: Time :math:`t`.
+      x_0: Sample from the source distribution.
+      x_1: Sample from the target distribution.
+
+    Returns:
+      Samples from the probability path between :math:`x_0` and :math:`x_1`
+        at time :math:`t`.
+    """
+    mu_t = self.compute_mu_t(t, x_0, x_1)
+    sigma_t = self.compute_sigma_t(t)
+    return mu_t + sigma_t * noise
+
+
+class StraightFlow(BaseFlow, abc.ABC):
+  """Base class for flows with straight paths."""
+
+  def compute_mu_t(
+      self, t: jnp.ndarray, x_0: jnp.ndarray, x_1: jnp.ndarray
+  ) -> jnp.ndarray:
+    """Compute the mean of the probablitiy path.
+
+      Compute the mean of the probablitiy path between :math:`x` and :math:`y`
+      at time :math:`t`.
+
+    Args:
+      t: Time :math:`t`.
+      x_0: Sample from the source distribution.
+      x_1: Sample from the target distribution.
+    """
+    return t * x_0 + (1 - t) * x_1
+
+  def compute_ut(
+      self, t: jnp.ndarray, x_0: jnp.ndarray, x_1: jnp.ndarray
+  ) -> jnp.ndarray:
+    """Evaluate the conditional vector field.
+
+    Evaluate the conditional vector field defined between :math:`x_0` and
+      :math:`x_1` at time :math:`t`.
+
+    Args:
+      t: Time :math:`t`.
+      x_0: Sample from the source distribution.
+      x_1: Sample from the target distribution.
+
+    Returns:
+      Conditional vector field evaluated at time :math:`t`.
+    """
+    return x_1 - x_0
+
+
+class ConstantNoiseFlow(StraightFlow):
+  r"""Flow with straight paths and constant flow noise :math:`\sigma`."""
+
+  def compute_sigma_t(self, t: jnp.ndarray):
+    r"""Compute noise of the flow at time :math:`t`.
+
+    Args:
+      t: Time :math:`t`.
+
+    Returns:
+      Constant, time-independent standard deviation :math:`\sigma`.
+    """
+    return self.sigma
+
+
+class BrownianNoiseFlow(StraightFlow):
+  r"""Brownian Bridge Flow.
+
+  Sampler for sampling noise implicitly defined by a Schroedinger Bridge
+  problem with parameter `\sigma` such that
+  :math:`\sigma_t = \sigma * \sqrt(t * (1-t))`.
+  """
+
+  def compute_sigma_t(self, t: jnp.ndarray):
+    """Compute the standard deviation of the probablity path at time :math:`t`.
+
+    Args:
+      t: Time :math:`t`.
+
+    Returns:
+      Standard deviation of the probablity path at time :math:`t`.
+    """
+    return jnp.sqrt(self.sigma * t * (1 - t))
+
+
+class BaseTimeSampler(abc.ABC):
```

Comment:
I think having classes for these samplers is unnecessarily complex at this point, would just define 2 functions with the correct signature (since they are just callables).